File: CreationalPattern/FactoryPattern/Python/Plants/PeaShooter.py

from Interface import Plant

class PeaShooter(Plant):
    """
        PeaShooter class
    """
    def __init__(self, *args, **kwargs):
        if args:
            super().__init__(args[0])
            self.price = args[1]
            self.health = args[2]
            self.damage = args[3]
        elif kwargs:
            super().__init__(kwargs.get('name','PeaShooter'))
            self.price = kwargs.get('price', 100)
            self.health = kwargs.get('health', 100)
            self.damage = kwargs.get('damage', 10)
        else:
            super().__init__('PeaShooter')
            self.price = 100
            self.health = 100
            self.damage = 10
    
    def show(self):
        print("Name: " + self.name 
                + " Price: " + str(self.price) 
                + " Health: " + str(self.health) 
                + " Damage: " + str(self.damage))

File: CreationalPattern/FactoryPattern/Java/Plants/PeaShooter.java

package Plants;

import Interface.Plant;

public class PeaShooter implements Plant {
    // Var parameters
    private String name;
    private int health;
    private int dame;
    private int price;
    // TODO parameters

    // Constructor
    public PeaShooter() {
        super();
        this.name = "PeaShooter";
        this.health = 100;
        this.dame = 10;
        this.price = 100;
    }

    public PeaShooter(String name, int health, int dame, int price) {
        super();
        this.name = name;
        this.health = health;
        this.price = price;
        this.dame = dame;
    }

    // Override method show
    @Override
    public void show() {
        System.out.println("Name: " + name);
        System.out.println("Health: " + health);
        System.out.println("Price: " + price);
        System.out.println("Dame: " + dame);
    }

    // Getter and Setter 
    // TODO add ...
}

